crates/bifrost/src/providers/replicated_loglet/loglet.rs

```diff
@@ -308,24 +379,53 @@ impl<T: TransportConnect> Loglet for ReplicatedLoglet<T> {
         Ok(())
     }
 
+    #[instrument(
+        level="error",
+        skip_all,
+        fields(
+            otel.name = "replicated_loglet: seal",
+        )
+    )]
     async fn seal(&self) -> Result<(), OperationError> {
-        let _ = SealTask::new(
-            self.my_params.clone(),
-            self.logservers_rpc.seal.clone(),
-            self.known_global_tail.clone(),
+        // lock-free fast-path
+        if self.known_global_tail.get().is_sealed() {
+            return Ok(());
+        }
+        trace!("seal() called");
+
+        // Ensure that only one seal operation is in progress at a time.
+        let start = Instant::now();
+        let _guard = self.seal_in_progress.lock().await;
+        trace!("seal() lock was released after {:?}", start.elapsed());
```

Comment:
"released" confused me a bit but I guess you are referring to the previous lock holder, right? Maybe acquired sets the subject to the caller.

Reply:
Good call.